### DISCUSSION TITLE
Multiplayer cross platform

### DISCUSSION THREAD

**Discussion by Galbi3000** [OpenRCT2/OpenRCT2#16897] [Q&A]
I am trying to get a Linux (Mint 21.10) laptop to run OpenRCT2 so that a Windows version can connect to it in multiplayer.

If I use the PPA for OpenRCT2 nightly builds the Windows version can not connect to it because of different versions. I have also tried the master PPA after doing some digging for more information. I have tried running the stable release version on Windows to see if it will connect to the Linux one using master, no joy. I have tried the latest development version, 2b371db, to see if it will connect to the nightly version on Linux. Nope!

So instead of using the version from the PPA I manually downloaded the latest development version OpenRCT2-0.3.5.1-develop-2b371db-linux-x86_64.tar.gz 
Running the manually downloaded version in its folder produces this error:
./openrct2: error while loading shared libraries: libicuuc.so.60: cannot open shared object file: No such file or directory

So, when I am able to run OpenRCT2 on my Linux machine I can not connect to a Windows version. When I make sure they are the same development version the Linux one does not work at all! What can I do?

I am trying to get this to work because I will have to spend a week away from home in a couple of days time and can only take my Linux laptop with me and I play OpenRCT2 with a friend who is using Windows.

**Comment by janisozaur** [comment]
'Nightly' builds are done each day on whatever the code on launchpad is - updated every few hours or at least daily. This will usually be around our current `develop` branch, which is where most of changes end up. We occasionally update the 'network version' to ensure clients communicate well. You must've gotten your build when launchpad still had a bit older version of code, while develop moved on. `openrct2 --version` should tell you versions of both your ends. The build you downloaded manually was produced on Ubuntu 20.04 and won't work due to mismatched versions of dependencies.

You can try waiting a bit for updated launchpad nightly build or use AppImage binary or build your own version.

**Reply by Galbi3000** [reply]
I ran OpenRCT2 with the --version on both Linux and Windows versions.

OpenRCT2 from the nightly build in Linux shows the following:
OpenRCT2, v0.3.5.1 (, DEBUG)
Linux (x86-64)
Network version: 0.3.5.1-17
Plugin API version: 45
Current park file version: 9
Minimum park file version: 9

Windows shows:
OpenRCT2, v0.3.5.1 (2b371db on develop) provided by GitHub
Windows (x86-64)
Network version: 0.3.5.1-23
Plugin API version: 51
Current park file version: 10
Minimum park file version: 9

As you can see, the network version is way behind on the Linux build! Also, the version number only shows as (, DEBUG) on the Linux build, which is true for nightly or master PPA installs.

Why does the manual download version only work with Ubuntu 20.04 when the versions in the PPA for 21.10 do?

Also, I ran the portable version of the stable release with the --version switch so see what it's network version is and all I got there was:
OpenRCT2, v0.3.5.1 (61c67af) provided by GitHub
Windows (x86-64)

So no way to tell.

This is all such a mess!!

**Reply by janisozaur** [reply]
> As you can see, the network version is way behind on the Linux build!

Indeed, I will check if launchpad needs any update.

> Also, the version number only shows as (, DEBUG) on the Linux build, which is true for nightly or master PPA installs.

You can check the package version instead.

> Why does the manual download version only work with Ubuntu 20.04 when the versions in the PPA for 21.10 do?

Explained this already

> no way to tell.

I have only added it past that release: https://github.com/OpenRCT2/OpenRCT2/pull/16413

> This is all such a mess

You're welcome to send PRs or provide [other means of unmessing it](https://github.com/sponsors/janisozaur/).

**Reply by janisozaur** [reply]
Turns out I filed that issue over a month ago and it wasn't picked up yet: https://github.com/OpenRCT2/OpenRCT2/issues/16691

**Reply by Galbi3000** [reply]
> > Why does the manual download version only work with Ubuntu 20.04 when the versions in the PPA for 21.10 do?
> 
> Explained this already

I'm sorry, not meaning to be a pain, but you did not explain this. What you wrote was: "The build you downloaded manually was produced on Ubuntu 20.04 and won't work due to mismatched versions of dependencies."
This does not explain why it works fine on 21.10 using the PPA version which is what I was asking in my response. 
I am no Linux expert so I find this a bit difficult to understand how the dependencies exist for the versions installed from the PPA but then the manually downloaded version can not use those same dependencies. Even if it's using the 20.04 version of those dependencies the newer ones would be backward compatible or else many other Linux programs might break as well. I know why 20.04 was used, because it's the current LTS release of Ubuntu, but it should also be stated clearly that this is the case.

Anyway, this does not solve the network compatibility issue. If I was able to use an older Windows build that uses network version Network version: 0.3.5.1-17 then I should not have any problems connecting to the Linux version. Is there a previous versions download page that goes that far back? That would have to be my short term fix unless they bring the Linux versions up to the same as Windows before I leave on Wednesday! lol

**Reply by janisozaur** [reply]
> (…) the manually downloaded version can not use those same dependencies

They are not the same. PPA build is done with versions from 21.10; the LTS build is done with those version of libraries.

> If I was able to use an older Windows build

You can find the builds in 'Actions' tab, they are stored for probably half of a year. You can also navigate to the commit/PR first and then go to checks for it. That said a preferred option would be to update your end, not downgrade the other.

**Reply by Galbi3000** [reply]
Yep that would the preferred option but I am not competent enough with Linux and OpenRCT2 to rebuild OpenRCT2 to use the latest network version. So my best option would be to find an old Windows version that uses the same network version as the PPA version uses! lol
Or rather, get my friend to download it and use that one. Or we can just not play it for the week I am away :/
No big deal in the short term but it has at least highlighted an issue with the builds that needs addressing.

**Reply by janisozaur** [reply]
Why don't you use AppImage then, at least until the launchpad issue is resolved?
